src/lib/y2users/user.rb

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast2/execute"
+
+module Y2Users
+  # Representing user configuration on system in contenxt of given User Configuration
+  # @note Immutable class.
+  class User
+    # @return [Y2Users::Configuration] reference to configuration in which it lives
+    attr_reader :configuration
+
+    # @return [String] user name
+    attr_reader :name
+
+    # @return [String, nil] user ID or nil if it is not yet assigned
+    attr_reader :uid
+
+    # @return [String, nil] primary group ID or nil if it is not yet assigned
+    # @note to get primary group use method #primary_group
+    attr_reader :gid
+
+    # @return [String, nil] default shell or nil if it is not yet assigned
+    attr_reader :shell
+
+    # @return [String, nil] home directory or nil if it is not yet assigned
+    attr_reader :home
+
+    # @return [Array<String>] Fields in GECOS entry
+    attr_reader :gecos
+
+    # @return [:local, :ldap, :unknown] where is user defined
+    attr_reader :source
+
+    # @see respective attributes for possible values
+    def initialize(configuration, name,
+      uid: nil, gid: nil, shell: nil, home: nil, gecos: [], source: :unknown)
```

Comment:
We have too much params here, but I prefer the method definition in just one line.

Perphaps changing `uid: nil, gid: nil, shell: nil, home: nil, gecos: [], source: :unknown` by `attrs = {}` and then using `@gecos = attrs[:gecos] || []` could help.

Reply:
Yes, this is something to improve. Having named params are useful to avoid passing invalid params. If we use a hash instead of named parameters, we should also add control for invalid values in the hash. But yes, we have to do something.